tools/qa/check_commits.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# HORTON: Helpful Open-source Research TOol for N-fermion systems.
+# Copyright (C) 2011-2016 The HORTON Development Team
+#
+# This file is part of HORTON.
+#
+# HORTON is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# HORTON is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+#
+# --
+"""Check commits for bad habits. Can be used as pre-commit when no arguments are given."""
+
+
+import argparse
+import subprocess
+import sys
+
+
+def main():
+    """Main program."""
+    ancestor = parse_args()
+    check_commits(ancestor)
+
+
+def parse_args():
+    """Determine which commits to check, from command-line args.
+
+    When no arguments are given, the index (commit to be made) will be checked. Otherwise,
+    all commits up to the ancestor are checked.
+    """
+    parser = argparse.ArgumentParser(description='Check basic formatting conventions in commits.')
+    parser.add_argument('ancestor', type=str, default=None, nargs='?',
```

Comment:
Should probably specify that ancestor should be a sha1 hash? It's obvious if you're familiar with git, but if you're not...